src/spikeinterface/extractors/neoextractors/neuroexplorer.py

```diff
@@ -0,0 +1,66 @@
+from pathlib import Path
+
+from spikeinterface.core.core_tools import define_function_from_class
+
+from .neobaseextractor import NeoBaseRecordingExtractor
+
+
+class NeuroExplorerRecordingExtractor(NeoBaseRecordingExtractor):
+    """
+    Class for reading NEX (NeuroExplorer data format) files.
+
+    Based on :py:class:`neo.rawio.NeuroExplorerRawIO`
+
+    Importantly, at the moment, this recorder only extracts one channel of the recording.
+    This is because the NeuroExplorerRawIO class does not support multi-channel recordings
+    as in the NeuroExplorer format they might have different sampling rates.
+
+    Consider exctracting all the channels and then concatenating them with the aggregate_channels function.
+
+    >>> from spikeinterface.extractors.neoextractors.neuroexplorer import NeuroExplorerRecordingExtractor
+    >>> from spikeinterface.core import aggregate_channels
+    >>>
+    >>> file_path="/the/path/to/your/nex/file.nex"
+    >>>
+    >>> streams = NeuroExplorerRecordingExtractor.get_streams(file_path=file_path)
+    >>> stream_names = streams[0]
+    >>>
+    >>> your_signal_stream_names = "Here goes the logic to filter from stream names the ones that you know have the same sampling rate and you want to aggregate"
+    >>>
+    >>> recording_list = [NeuroExplorerRecordingExtractor(file_path=file_path, stream_name=stream_name) for stream_name in your_signal_stream_names]
+    >>> recording = aggregate_channels(recording_list)
+
+
+
+    Parameters
+    ----------
+    file_path: str
+        The file path to load the recordings from.
+    stream_id: str, optional
+        If there are several streams, specify the stream id you want to load.
+        For this neo reader streams are defined by their sampling frequency.
+    stream_name: str, optional
+        If there are several streams, specify the stream name you want to load.
+    all_annotations: bool, default: False
```

Comment:
And a question... What format did you want
`Optional[str]`,` str, optional,` or `str, default: None`.  Just so I know if I proofread docstrings in the future. @h-mayorquin @alejoe91

Reply:
I don't have a strong view on this. I think that me and Alessio use the second option because a vscode produces that by default. It seems that numpy does the same thing:
https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html
https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html

I kind of wish there was just some standard but those things are not fully specified as far as I know.
Do you have preference?

Reply:
Honestly the one I like the most is the third option: `str, default: None`. I know that this is very far from consistent in the code-base, but I think as we move forward it's very good to make decisions also on these apparently small things :)

Reply:
Yeah I tend to switch depending on context. When I'm editing docstrings I switch it to the third option, but when I'm writing a docstring I just let it autoformat. So honestly I don't care the actual format as much as I care about trying to make a code base consistent.

Reply:
Maybe we can add this to the developer guide, at least to freeze the preference and slowly move in that direction.

https://spikeinterface.readthedocs.io/en/latest/development/development.html

Reply:
good point. Let's make a PR